contracts/starknet/space/space.cairo

```diff
@@ -75,25 +85,36 @@ end
 @constructor
 func constructor{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr : felt}(
         _voting_delay : felt, _voting_period : felt, _proposal_threshold : Uint256,
-        _voting_strategy : felt, _authenticator : felt):
+        _voting_strategy : felt, _authenticator : felt, _l1_executor : felt):
     # Sanity checks
     assert_nn(_voting_delay)
     assert_nn(_voting_period)
     assert_not_zero(_voting_strategy)
     assert_not_zero(_authenticator)
     # TODO: maybe use uint256_signed_nn to check proposal_threshold?
+    # TODO: maybe check that _l1_executor is not 0?
 
     # Initialize the storage variables
     voting_delay.write(_voting_delay)
     voting_period.write(_voting_period)
     proposal_threshold.write(_proposal_threshold)
     voting_strategy.write(_voting_strategy)
     authenticator.write(_authenticator)
+    l1_executor.write(EthAddress(_l1_executor))
     next_proposal_nonce.write(1)
 
     return ()
 end
 
+# TODO: L1 needs to know about L2 address, but L2 needs to know about the L2 address... need to fix that.
+# TODO: this should either be on the l1 contract or the l2 contract. Since l1 contract has an `owner` I think it should be on l1.
```

Comment:
This needs to be addressed (we can merge like this, but will definitely need to be addressed soon)